### PR TITLE
Move DbgCtl.cc from tscore to tsapicore

### PR DESCRIPTION
**Pull request by cmcfarlen**

### CODE REVIEW

**Comment by ywkaras** [comment]
I think, for this to work, we would also have to turn tsapicore into a dynamically linked lib.

**Review by dragon512 — APPROVED** [review]

**Review by dragon512 — COMMENTED** [review]
For the cmake part of this

**Comment by cmcfarlen** [comment]
> I think, for this to work, we would also have to turn tsapicore into a dynamically linked lib.

I tried this but there is a dependency cycle, so we will have to move some items around.  We need to do this anyway.

File: iocore/net/Makefile.am [hunk 1]
```diff
@@ -53,6 +53,7 @@ test_certlookup_SOURCES = \
 test_certlookup_LDADD = \
 	$(top_builddir)/iocore/eventsystem/libinkevent.a \
 	$(top_builddir)/src/tscore/libtscore.a \
+	$(top_builddir)/src/api/libtsapicore.a \
```
Comment (ywkaras) [inline iocore/net/Makefile.am:56]: I'm pretty sure tsapicore.a need to become dynamically loaded and installed.  @dragon512 can confirm.

**Review by bneradt — APPROVED** [review]
Looks great. Thanks for making this happen.

**Pull request merged**